### ISSUE TITLE
aperturescience.com: DefineVideoStream, got Invalid data: Invalid video codec.

### ISSUE THREAD

**Issue by FoxMcCloud45** [ruffle-rs/ruffle#5704]
### Describe the bug

https://www.aperturescience.com uses an old SWF file made in 2010.

In Mozilla Firefox, the first frame of the video shows but nothing else happens.
In Google Chrome, the Flash window shows a PLAY button, clicking it causes a black render.

The website's SWF file is hosted at https://www.aperturescience.com/aperturescience.swf

Also provided below.
[aperturescience.zip](https://github.com/ruffle-rs/ruffle/files/7584238/aperturescience.zip)
 .

### Expected behavior

The SWF's video should play normally.

### Affected platform

Browser's extension

### Operating system

Microsoft Windows 10 LTSC 2019 (1809)

### Browser

Mozilla Firefox 91.3.0esr

### Additional information

Firefox's JavaScript console log is attached below.
[console-export-2021-11-22_21-27-39.txt](https://github.com/ruffle-rs/ruffle/files/7584230/console-export-2021-11-22_21-27-39.txt)

**Comment by n0samu** [comment]
This SWF uses an external video file. Ruffle doesn't support loading external video/audio/image assets yet.

**Comment by vladislavkoz** [comment]
Hello guys. I have SWF which uses some FLV video files. As I understand support for FLV is not added yet. And as I understand, we can't use any video files. Is it correct? In my case, I have the following error: 
![image](https://user-images.githubusercontent.com/40685761/160382402-d8cb1289-1dec-4840-9e4f-f1cd1611b3ab.png)

Is there any scheduled work on adding video files support then? Could you propose any workaround for this before video support is added?

**Comment by danielhjacobs** [comment]
Embedded H.263 and VP6(A) videos work. FLV's/external videos do not work.

**Comment by vladislavkoz** [comment]
> A)

So does it mean that we can't use external videos at all?

**Comment by danielhjacobs** [comment]
Correct. You need to embed an H.263 or VP6 video in the swf, like this: https://dnalc.cshl.edu/view/962-Causes-Smoking-Prevention.html

**Comment by Lord-McSweeney** [comment]
Video has been implemented- however, this seems to need something else to work, as it currently just shows a single frame. Stub warning: `Stub: Stream data processing (name: onXMPData)`

**Comment by n0samu** [comment]
As I mentioned in https://github.com/ruffle-rs/ruffle/pull/12091#issuecomment-1674177630:
> The video shows only the first frame, and more strangely, the audio plays but periodically glitches out. Might there be some script data interleaved with the audio that Ruffle is erroneously trying to play back?

**Comment by torokati44** [comment]
This SWF streams two FLV files:
https://www.aperturescience.com/wcc_holiday10_v03.flv
https://www.aperturescience.com/wcc_holiday10_v03_egg.flv

They each have a VP6 video in them, which is well supported.

And I can confirm that they are streamed and decoded correctly - they just aren't visible on the stage for some reason!

**Comment by torokati44** [comment]
Ah, `_level0.cam1._videoHolder` has its visibility turned off (or rather, not turned on) for some reason. Flipping it on manually does show the video playing. Although, at a weird position/size/aspect ratio:

![image](https://github.com/ruffle-rs/ruffle/assets/288816/53ecf26e-33a6-4c37-b568-2c1b1e8437e9)

**Comment by shaunroselt** [comment]
+1

**Comment by Lord-McSweeney** [comment]
Closing in favor of #20606 as this issue was back from when video files didn't work at all
